apps/docs/docs/core/infer-types.md

```diff
@@ -0,0 +1,69 @@
+# Inferring Types
+
+Often, we need to manually extract the request or responses types of specific contract endpoints, so functions, services, lambdas, React components, etc. can be safely typed
+when there is no automatic type inference.
+
+We have separate type helpers for server-side and client-side code since we need to infer either the Input or Output Zod types depending on
+where the code is used.
+
+## Inferring Response Types
+
+To get the response types of a contract or a specific endpoint, we have the following type helpers:
+
+- `InferResponsesForServer<AppRouter | AppRoute, OptionalHttpStatusCode>`
+- `InferResponsesForClient<AppRouter | AppRoute, OptionalHttpStatusCode>`
+
+```typescript
+import { InferResponsesForServer } from '@ts-rest/core';
+import { contract } from './contract';
+
+type ResponseShapes = InferResponsesForServer<typeof contract>;
+
+async function someHttpCall(req: Request): Promise<ResponseShapes['getPosts']> {
+  return ...;
+}
+
+function someServiceCall(): InferResponsesForServer<typeof contract.getPosts> {
+  return ...;
+}
+```
+
+### Inferring Response Body
+
+If you need to infer the response body for a defined response status of a specific endpoint, we can use the following type helpers:
+
+- `InferResponseBodyForServer<AppRoute, OptionalHttpStatusCode>`
+- `InferResponseBodyForClient<AppRoute, OptionalHttpStatusCode>`
+
+This is syntactic sugar for `InferResponsesForServer<AppRoute, OptionalHttpStatusCode & keyof AppRoute['responses']>['body']`
+
+```typescript
+import React from 'react';
+import { InferResponseBodyForClient } from '@ts-rest/core';
+import { contract } from './contract';
+
+type Post = InferResponseBodyForClient<typeof contract.getPost, 200>;
```

Comment:
Instead of `InferResponseBodyForClient`, why not add another type generic param that allows the dev to pass in `server` or `client` and rename to `InferResponseBody`? 

So it would look like this:
```suggestion
type Post = InferResponseBody<typeof contract.getPost, 200, 'server'>;
```

Reply:
@michaelangrivera For the code on our end, yes, it would help a bit with code reuse. I think from a developers perspective though that it's safer to have separate types with `Client` or `Server` in the type name; it'll be clearer in terms of readability and reduces the chances of copy-pasting errors from being missed. To hammer on that point, I actually changed it now to `ServerInfer` and `ClientInfer`, so now it's shorter (I was also annoyed with `ForClient`/`ForServer`) and also much more readable.